xray/backends/pynio_.py

```diff
@@ -0,0 +1,47 @@
+import numpy as np
+
+from .. import Variable
+from ..core.utils import FrozenOrderedDict, Frozen, NDArrayMixin
+from ..core import indexing
+
+from .common import AbstractDataStore
+
+
+class NioArrayWrapper(NDArrayMixin):
+    def __init__(self, array, ds):
+        self.array = array
+        self._ds = ds  # make an explicit reference because pynio uses weakrefs
```

Comment:
This tripped me up a little bit -- I think it would be better to use normal references from the Variables back to Files.